li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/common/LargeMessageHeaderValue.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.common;
+
+import com.linkedin.kafka.clients.utils.PrimitiveEncoderDecoder;
+import java.util.UUID;
+
+
+/**
+ * This class represents the header value for a large message.
+ * Every large message header takes up 25 bytes and is structured as follows
+ *
+ * | Type   | UUID     | segmentNumber | numberOfSegments |
+ * | 1 byte | 16 bytes | 4 bytes       | 4 bytes          |
+ *
+ * The Large message header values will be used to support large messages eventually.
+ * (as opposed to encoding large segment metadata info inside the payload)
+ */
+public class LargeMessageHeaderValue {
+  public static final UUID EMPTY_UUID = new UUID(0L, 0L);
+
+  private final byte _type;
+  private final UUID _uuid;
+  private final int _segmentNumber;
+  private final int _numberOfSegments;
+
+  // This indicates that the large message framework is using
+  // SegmentSerializer/SegmentDeserializer interface to split
+  // and assemble large message segments.
+  public static final byte LEGACY = (byte) 0;
+
+  public LargeMessageHeaderValue(byte type, UUID uuid, int segmentNumber, int numberOfSegments) {
+    _type = type;
+    _uuid = uuid;
+    _segmentNumber = segmentNumber;
+    _numberOfSegments = numberOfSegments;
+  }
+
+  public int getSegmentNumber() {
+    return _segmentNumber;
+  }
+
+  public int getNumberOfSegments() {
+    return _numberOfSegments;
+  }
+
+  public UUID getUuid() {
+    return _uuid;
+  }
+
+  public byte getType() {
+    return _type;
+  }
+
+  public static byte[] toBytes(LargeMessageHeaderValue largeMessageHeaderValue) {
+    // 1 byte for type
+    // 16 bytes for UUID
+    // 4 bytes for segment number
+    // 4 bytes for number of segments
+    byte[] serialized = new byte[25];
+    serialized[0] = largeMessageHeaderValue.getType();
+    PrimitiveEncoderDecoder.encodeLong(largeMessageHeaderValue.getUuid().getLeastSignificantBits(), serialized, 1);
+    PrimitiveEncoderDecoder.encodeLong(largeMessageHeaderValue.getUuid().getMostSignificantBits(), serialized, 1 + PrimitiveEncoderDecoder.LONG_SIZE);
+    PrimitiveEncoderDecoder.encodeInt(
+        largeMessageHeaderValue.getSegmentNumber(),
+        serialized,
+        1 + 2 * PrimitiveEncoderDecoder.LONG_SIZE
+    );
+    PrimitiveEncoderDecoder.encodeInt(
+        largeMessageHeaderValue.getNumberOfSegments(),
+        serialized,
+        1 + 2 * PrimitiveEncoderDecoder.LONG_SIZE + PrimitiveEncoderDecoder.INT_SIZE
+    );
+    return serialized;
+  }
+
+  public static LargeMessageHeaderValue fromBytes(byte[] bytes) {
+    byte type = bytes[0];
+    long leastSignificantBits = PrimitiveEncoderDecoder.decodeLong(bytes, 1);
```

Comment:
nit: using a variable to keep track of the current offset in the byte array makes it easier to read the code.

```
int bytesOffset = 0;
bytesOffset += 1; // for lsb of UUID
bytesOffset += PrimitiveEncoderDecoder.LONG_SIZE; // for msb of UUID
bytesOffset += PrimitiveEncoderDecoder.LONG_SIZE; // for segment number
bytesOffset += PrimitiveEncoderDecoder.INT_SIZE; // for number of segments
```